src/component/tooltip/TooltipHTMLContent.ts

```diff
@@ -375,20 +375,26 @@ class TooltipHTMLContent {
             return;
         }
 
+        const el = this.el;
+
         if (isString(arrowPosition) && tooltipModel.get('trigger') === 'item'
             && !shouldTooltipConfine(tooltipModel)) {
             content += assembleArrow(tooltipModel.get('backgroundColor'), borderColor, arrowPosition);
         }
-        if (isObject(content)) {
-            if (this.el.children) {
-                for (var child of Array.from(this.el.children)) {
-                    this.el.removeChild(child);
+        if (isString(content)) {
+            el.innerHTML = content == null ? '' : content;
+        }
+        else if (content) {
+            // Clear previous
+            el.innerHTML = '';
+            if (!isArray(content)) {
+                content = [content];
+            }
+            for (let i = 0; i < content.length; i++) {
```

Comment:
Thanks for cleaning this up!
I'm curious, is a `for` loop with repeated indexing better than a `for … of` loop? I know that it is more memory efficient in compiled languages where arrays are implemented as contiguous areas of memory, but I had no idea the same was true for Javascript.